ui/v2.5/src/components/Shared/StashBoxIDSearchModal.tsx

```diff
@@ -132,8 +144,121 @@ export const PerformerSearchResult: React.FC<IPerformerResultProps> = ({
   );
 };
 
+// Scene Result Component
+interface ISceneResultProps {
+  scene: GQL.ScrapedSceneDataFragment;
+}
+
+const SceneSearchResultDetails: React.FC<ISceneResultProps> = ({ scene }) => {
+  return (
+    <div className="scene-result">
+      <Row>
+        <SearchResultImage imageUrl={scene.image} />
+        <div className="col flex-column">
+          <h4 className="scene-title">
+            <span>{scene.title}</span>
+            {scene.code && (
+              <span className="scene-code">{` (${scene.code})`}</span>
+            )}
+          </h4>
+          <h5 className="scene-details">
+            {scene.studio?.name && <span>{scene.studio.name}</span>}
+            {scene.date && (
+              <span className="scene-date">{` • ${scene.date}`}</span>
+            )}
+          </h5>
+          {scene.performers && scene.performers.length > 0 && (
+            <div className="scene-performers">
+              {scene.performers.map((p) => p.name).join(", ")}
+            </div>
+          )}
+        </div>
+      </Row>
+      <Row>
+        <Col>
+          <TruncatedText text={scene.details ?? ""} lineCount={3} />
+        </Col>
+      </Row>
+      <SearchResultTags tags={scene.tags} />
+    </div>
+  );
+};
+
+export const SceneSearchResult: React.FC<ISceneResultProps> = ({ scene }) => {
+  return (
+    <div className="mt-3 search-item" style={{ cursor: "pointer" }}>
+      <SceneSearchResultDetails scene={scene} />
+    </div>
+  );
+};
+
+// Studio Result Component
+interface IStudioResultProps {
+  studio: GQL.ScrapedStudioDataFragment;
+}
+
+const StudioSearchResultDetails: React.FC<IStudioResultProps> = ({
+  studio,
+}) => {
+  return (
+    <div className="studio-result">
+      <Row>
+        <SearchResultImage imageUrl={studio.image} />
+        <div className="col flex-column">
+          <h4 className="studio-name">
+            <span>{studio.name}</span>
+          </h4>
+          {studio.parent?.name && (
+            <h5 className="studio-parent">
+              <span>{studio.parent.name}</span>
+            </h5>
+          )}
+          {studio.urls && studio.urls.length > 0 && (
+            <div className="studio-url text-muted small">{studio.urls[0]}</div>
+          )}
+        </div>
+      </Row>
+    </div>
+  );
+};
+
+export const StudioSearchResult: React.FC<IStudioResultProps> = ({
+  studio,
+}) => {
+  return (
+    <div className="mt-3 search-item" style={{ cursor: "pointer" }}>
+      <StudioSearchResultDetails studio={studio} />
+    </div>
+  );
+};
+
+// Helper to get entity type display name for i18n
+function getEntityTypeDisplayName(entityType: StashBoxEntityType): string {
+  switch (entityType) {
+    case "performer":
+      return "Performer";
+    case "scene":
+      return "Scene";
+    case "studio":
+      return "Studio";
+  }
+}
```

Comment:
This needs to be internationalised as well. This should return a message id, which would be translated using `intl` in the calling code.